src/meta/src/hummock/manager/mod.rs

```diff
@@ -2339,58 +2350,70 @@ where
             for (table_id, table_size) in &group.table_statistic {
                 let mut is_high_write_throughput = false;
                 let mut is_low_write_throughput = true;
+                let mut do_not_split = true;
                 if let Some(history) = table_write_throughput.get(table_id) {
                     if history.len() >= HISTORY_TABLE_INFO_WINDOW_SIZE {
-                        let window_total_size = history.iter().sum::<u64>();
                         is_high_write_throughput = history.iter().all(|throughput| {
                             *throughput > self.env.opts.table_write_throughput_threshold
                         });
-                        is_low_write_throughput = window_total_size
-                            < (HISTORY_TABLE_INFO_WINDOW_SIZE as u64)
-                                * self.env.opts.min_table_split_write_throughput;
+                        is_low_write_throughput = history.iter().any(|throughput| {
+                            *throughput < self.env.opts.min_table_split_write_throughput
```

Comment:
Should we take checkpoint interval into consideration when calculate the throughput? barrier_interval_ms and checkpoint_frequency can be changed in runtime via SQL.

Reply:
The interval is not proportional to the amount of data, so it is difficult to calculate the throughput based on the interval.

+1 for changing in runtime via SQL.

Reply:
I don't mean we need to expose it via SQL in my original comment (although it is a good idea). 

I mean since `barrier_interval_ms` and `checkpoint_frequency` can be changed at any time, we should divide the throughput calculated here either by `barrier_interval_ms * checkpoint_frequency` or more accurately by system time.

Reply:
Agree. I have refactor it